File: app/src/main/java/home/oleg/placesnearme/common/provider/ResourceProviderImpl.java
package home.oleg.placesnearme.common.provider;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;

import javax.inject.Inject;

/**
 * Created by Oleg Sheliakin on 28.08.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public class ResourceProviderImpl implements ResourceProvider {

    private final Context context;

    @Inject
    public ResourceProviderImpl(Context context) {
        this.context = context;
    }

    @Override
    public Drawable getDrawable(int resId) {
        return ContextCompat.getDrawable(context, resId);
    }

    @Override
    public String getString(int resId) {
        return context.getString(resId);
    }

    @Override
    public int getColor(int resId) {
        return ContextCompat.getColor(context, resId);
    }
}

File: core-tools/src/main/java/com/home/olegsheliakin/corettools/resource/ResourceProviderImpl.kt
package com.home.olegsheliakin.corettools.resource

import android.content.Context
import android.graphics.drawable.Drawable

import androidx.core.content.ContextCompat

import javax.inject.Inject

/**
 * Created by Oleg Sheliakin on 28.08.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
class ResourceProviderImpl @Inject
constructor(private val context: Context) : ResourceProvider {

    override fun getDrawable(resId: Int): Drawable? {
        return ContextCompat.getDrawable(context, resId)
    }

    override fun getString(resId: Int): String {
        return context.getString(resId)
    }

    override fun getColor(resId: Int): Int {
        return ContextCompat.getColor(context, resId)
    }
}
